### PR TITLE
Add Haiku support.

### PR DESCRIPTION
**Pull request by korli**
Add include of sys/select.h to get FD_* definitions.

### CODE REVIEW

**Comment by jonesmz** [comment]
Is the modification to configure.py useful or necessary?

**Comment by jonesmz** [comment]
No cmake changes needed?

**Comment by korli** [comment]
the modification in configure.py is needed for --platform: https://github.com/ninja-build/ninja/blob/a30ccaef7314f5aa858ba3167af08be4b518d960/configure.py#L208

**Comment by korli** [comment]
> No cmake changes needed?

AFAIK we don't build ninja with cmake on Haiku. Should we?

**Comment by jonesmz** [comment]
cmake is preferred over configure.py. No one should be using configure.py for anything.

**Comment by korli** [comment]
> cmake is preferred over configure.py. No one should be using configure.py for anything.

Well, I saw ArchLinux using configure.py :)

I tested with cmake and it builds out of the box, nice. Should I clean up the pull request, removing configure.py changes?

**Comment by jonesmz** [comment]
Personally i think you should, but I am not a Ninja maintainer, so do what you think is best.

File: src/subprocess-posix.cc [hunk 1]
```diff
@@ -21,6 +21,7 @@
 #include <unistd.h>
 #include <stdio.h>
 #include <string.h>
+#include <sys/select.h>
```
Comment (jhasse) [inline src/subprocess-posix.cc:24]: why?

Comment (korli) [reply src/subprocess-posix.cc:24]: to get FD_* definitions from sys/select.h

https://github.com/ninja-build/ninja/blob/master/src/subprocess-posix.cc#L334
https://pubs.opengroup.org/onlinepubs/007904975/basedefs/sys/select.h.html

Comment (korli) [reply src/subprocess-posix.cc]: I checked again, and it seems to work fine without it. Must be a restover. It's now removed.

**Pull request merged**